protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/record/value/deployment/DeploymentRecord.java

```diff
@@ -27,8 +29,17 @@
   private final ArrayProperty<ProcessMetadata> processesMetadataProp =
       new ArrayProperty<>(PROCESSES, new ProcessMetadata());
 
+  private final ArrayProperty<DecisionRecord> decisionMetadataProp =
+      new ArrayProperty<>("decisionsMetadata", new DecisionRecord());
+
+  private final ArrayProperty<DecisionRequirementsMetadataRecord> decisionRequirementsMetadataProp =
+      new ArrayProperty<>("decisionRequirementsMetadata", new DecisionRequirementsMetadataRecord());
+
   public DeploymentRecord() {
-    declareProperty(resourcesProp).declareProperty(processesMetadataProp);
+    declareProperty(resourcesProp)
+        .declareProperty(processesMetadataProp)
+        .declareProperty(decisionRequirementsMetadataProp)
+        .declareProperty(decisionMetadataProp);
```

Comment:
❓ Can we simply declare these properties? What about existing records without these properties? Is that accounted for?

Reply:
This is the way :sweat_smile: 

The default value of an array property is an empty array. If we read an old record then the record can be deserialized and the property is empty. So, this should be fine :+1:

Reply:
You hope :grin: :sweat_smile: I think this only works if it is added to the end :D

Reply:
I ran a unit test to verify my assumption. The records are encoded in message pack. The properties are independent from the order 😉